File: dashboard/src/api/tasks.ts
import { apiClient } from "./client";
import type { TaskDTO } from "@/types";

export function createTask(issueId: number, assignedTo: string): Promise<TaskDTO> {
  return apiClient.post<TaskDTO>("/tasks", { issue_id: issueId, assigned_to: assignedTo });
}

export function updateTaskStatus(taskId: number, status: string): Promise<TaskDTO> {
  return apiClient.patch<TaskDTO>(`/tasks/${taskId}`, { status });
}

File: backend/app/api/v1/tasks.py
from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session

from app.core.security import get_current_user
from app.db.session import get_db
from app.models.task import Task
from app.schemas.task import TaskCreate, TaskOut, TaskStatusUpdate
from app.services import issues as issues_service
from app.services.tasks import assignment as tasks_service

router = APIRouter(prefix="/tasks", tags=["tasks"], dependencies=[Depends(get_current_user)])


@router.post("", response_model=TaskOut, status_code=status.HTTP_201_CREATED)
def create_task(payload: TaskCreate, db: Session = Depends(get_db)):
    issue = issues_service.get_issue(db, payload.issue_id)
    if issue is None:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Issue not found")
    return tasks_service.create_task(db, issue, payload.assigned_to)


@router.patch("/{task_id}", response_model=TaskOut)
def update_task_status(task_id: int, payload: TaskStatusUpdate, db: Session = Depends(get_db)):
    task = db.get(Task, task_id)
    if task is None:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Task not found")
    return tasks_service.update_task_status(db, task, payload.status)
